### ISSUE TITLE
Bring XBee out of tree driver into mainline

### ISSUE THREAD

**Issue by Stefan-Schmidt** [linux-wpan/linux-wpan#1]
There exists an out of tree driver for the XBee modules. Having support for it in mainline would be beneficial.

https://github.com/joaopedrotaveira/linux-rpl/blob/master/mainline-3.12.y/0001-Added-XBee-driver-support.patch

**Comment by alexaring** [comment]
The XBee module is also interesting for supporting the first HardMAC driver.
This transceiver use some AT commands to trigger mlme-ops, setup own "llsec" implementation on transceiver side.

Some note about "mac security handling on HardMAC drivers". There exist currently a "architecture issue". Wireshark see all "RAW" mac802154 frames encrypted. If we support such things on HardMAC side then the frame can't (should not) be encrypted before reach driver-layer.

It also can be used for testing mlme-ops.

**Comment by Stefan-Schmidt** [comment]
Agreed, support for this would be nice for having a HardMAC driver.
The device has an AT mode as well as an API mode. The later seems to make the communication to the hardware a bit easier from a driver perspective compared to AT commands which are easy to use manually. The driver seem to use the API interface already.
